### ISSUE TITLE
Respect local includes in remote files

### ISSUE THREAD

**Issue by deldrid1** [electricimp/Builder#42]
Say I have 3 files in the same repo, `src.nut`, `lib/lib.nut`, and `lib/libDependency.nut`

In `src.nut`
```squirrel
...
@include once "lib/lib.nut"
...
```

and in `lib.nut`
```squirrel
...
@include once "./libDependency.nut"
...
```

If I use builder for this example, life is good and everything works as expected.  However, if I want to include lib from **another** repo, things begin to fall apart:

```squirrel
 ‘@include once "github:someOwner/someUser/lib/lib.nut@develop"
```
> fails with an ‘Error: Local file "lib/libDependency.nut" not found (otherRepo.nut:LineNumber)’. 

My proposed approach would be a modification to the way that remote file @include’s are handled.  I’m not intimately familiar with how builder is architected, but basically, each time a remote file or github file was included, that remote path or github path/branch would need to be stored in memory along with the source code.  Any time a local include is found inside of these remote files, it would prepend the remote path and include the locally referenced file at the remote path.

**Comment by deldrid1** [comment]
@ppetrosh - any updates here?  This one continues to bite...

**Comment by ppetrosh** [comment]
This one is actually a major and a breaking change. And it's not specific to local or remote resources. It's about switching to relevant paths from absolute ones. So if we do the change, I think it should work this way even for local resources not only for remote ones. But let me think about it some more.

**Comment by ragrus-nbl** [comment]
@deldrid1 We can add an option to allow interpreting as relative instead of absolute all the paths of `@include` directives located in the files included from GitHub. If you use this option, `@include`'s will be handled the way you described above. Will this work for you? Is it enough for you to handle only GitHub `@include`'s?

**Comment by deldrid1** [comment]
GitHub is the big one that my team uses so I think for now the answer is yes.

**Comment by betzrhodes** [comment]
Fix for this issue just released in [v2.8.0](https://github.com/electricimp/Builder/tree/v2.8.0)

- Add --use-remote-relative-includes CLI option.
  - Fixed issues: #42
